pkg/csi/service/node.go

```diff
@@ -362,23 +353,90 @@ func (s *service) NodeGetInfo(
 	ctx context.Context,
 	req *csi.NodeGetInfoRequest) (
 	*csi.NodeGetInfoResponse, error) {
-
-	id, err := os.Hostname()
+	nodeID, err := os.Hostname()
 	if err != nil {
 		return nil, status.Errorf(codes.Internal,
 			"Unable to retrieve Node ID, err: %s", err)
 	}
+	var cfg *cnsconfig.Config
+	cfgPath = csictx.Getenv(ctx, cnsconfig.EnvCloudConfig)
+	if cfgPath == "" {
+		cfgPath = cnsconfig.DefaultCloudConfigPath
+	}
+	cfg, err = cnsconfig.GetCnsconfig(cfgPath)
+	if err != nil {
+		klog.Errorf("Failed to read cnsconfig. Error: %v", err)
+		return nil, status.Errorf(codes.Internal, err.Error())
+	}
+	var accessibleTopology map[string]string
+	topology := &csi.Topology{}
+
+	if cfg.Labels.Zone != "" && cfg.Labels.Region != "" {
```

Comment:
Okay, so to answer my question from above the nodes need access to vCenter in order to figure out topology. I guess there really is no way around that...